### PR TITLE
PUBDEV-4802: max_runtime_secs too small.  Make sure all algos return …

### PR DESCRIPTION
**Pull request by wendycwong**
Two things:

1. Added code to different algos to make sure that they return a valid model even if the max_runtime_secs is too short.
2. Added pyunit_test to make sure it works.

### CODE REVIEW

File: h2o-algos/src/main/java/hex/deeplearning/DeepLearning.java [hunk 1]
```diff
@@ -479,6 +479,8 @@ public final DeepLearningModel trainModel(DeepLearningModel model) {
         if (!_parms._quiet_mode) {
           Log.info("==============================================================================================================================================================================");
           if (stop_requested()) {
+            warn("_max_runtime_secs", "Deep Learning model training was interrupted.  " +
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/deeplearning/DeepLearning.java]: This can also happen when the user cancels the job explicitly. We should probably only show this warning if there was a timeout (timeout() function returns true).

File: h2o-algos/src/main/java/hex/svd/SVD.java [hunk 2]
```diff
@@ -701,9 +706,12 @@ public void computeImpl() {
           }
 
           for (int k = 1; k < _parms._nv; k++) {  // loop through for each eigenvalue/eigenvector...
-            if (stop_requested()) break;
-            if (_matrixRankReached) { // number of eigenvalues found is less than _nv
-              int newk = k-1;
+            if (_matrixRankReached || stop_requested()) { // number of eigenvalues found is less than _nv
+              if (stop_requested()) {
+                _matrixRankReached = true;
```
Comment (michalkurka) [inline h2o-algos/src/main/java/hex/svd/SVD.java]: Can you please explain why is this needed?

Comment (wendycwong) [inline h2o-algos/src/main/java/hex/svd/SVD.java]: This is needed because:

1. Some datasets are rank deficient.  Hence, a dataset is of rank 10 say and a user is asking for k = 12 which is 2 more than what the dataset's true rank is.  The PCA can detect this and will therefore set the k to be 10 instead of 12.  In this case, a warning will be sent back to the client to let them know this.
2. Due to max_runtime_secs restriction, we may not have time to find the number of eigenvalues/eigenvectors specified by the user in parms._nv.  Hence, in this case again, we will need to reduce the parameter to be the actual number of eigenvalue/eigenvector we found instead of the ones asked by the user.

Comment (michalkurka) [inline h2o-algos/src/main/java/hex/svd/SVD.java]: Thanks Wendy.

The issue I see with setting the parameter to true is that it doesn't actually capture the property of the dataset: we don't now if it is rank deficient, we stopped early because of the timeout. Setting this parameter to true is therefore a little confusing.

If I read the code correctly the only part that depends on it is this:

        if (_matrixRankReached) { // need to shorten the correct eigen stuff
          model._output._d = Arrays.copyOf(model._output._d, _parms._nv);
          for (int index=0; index < model._output._v.length; index++) {
            model._output._v[index] = Arrays.copyOf(model._output._v[index],  _parms._nv);
          }
        }

Can we change it simply to

        if (_parms._nv < model._output._d.length) { // need to shorten the correct eigen stuff
          model._output._d = Arrays.copyOf(model._output._d, _parms._nv);
          for (int index=0; index < model._output._v.length; index++) {
            model._output._v[index] = Arrays.copyOf(model._output._v[index],  _parms._nv);
          }
        }

and keep the "_matrixRankReached" unmodified.

We can also change the name of "_matrixRankReached" to something else. My point is that it should correctly describe the state of things. If we stop early the rank was not actually reached.

Comment (wendycwong) [inline h2o-algos/src/main/java/hex/svd/SVD.java]: Agreed.  That is good practiced.  Done.
